src/templates_front/templator_front.org

```diff
@@ -958,11 +958,11 @@ trexio_open(const char* file_name, const char mode,
 
    #+begin_src f90 :tangle prefix_fortran.f90
 interface
-   integer(8) function trexio_open_c (filename, mode, backend, rc_open) bind(C, name="trexio_open")
+   integer(c_long) function trexio_open_c (filename, mode, backend, rc_open) bind(C, name="trexio_open")
```

Comment:
Let me understand this a little bit better: in the C API we had to explicitly use integer types `stdint.h`, e.g. `int64_t`, to avoid issues with the `long int` being 4-byte on 32-bit systems and 8-byte on 64-bit systems. As far as I understand, using `integer(c_long)` as a replacement of `integer(8)` might resurrect the aforementioned issues. 

Perhaps we should use explicit `integer(c_int64_t)` instead of `integer(c_long)` (and also `c_int64_t` instead of `c_int`). I found these constants on the [GCC website](https://gcc.gnu.org/onlinedocs/gcc-4.9.0/gfortran/ISO_005fC_005fBINDING.html#ISO_005fC_005fBINDING).

@scemama @plopezrios What would you say?

Reply:
Sorry, I missed that one!
We use `int32_t` and `int64_t` which are exactly `integer(4)` and `integer(8)`. `c_int` and `c_long` correspond to conventional `int` and `long` types which can differ among architectures. So `integer(8)` is correct.

Reply:
I made these changes to avoid `-Wc-binding-type` warnings.  Suppressing these requires declaring all variables in interfaces with the `bind(C)` attribute to be declared using kinds from the iso_c_binding module.  c_int64_t and c_int32_t should work better than my initial c_long and c_int.